arangod/Replication/ReplicationClients.cpp

```diff
@@ -34,12 +34,39 @@
 
 using namespace arangodb;
 
+namespace arangodb {
+// Helper for logging
+
+struct SyncerInfo {
+  explicit SyncerInfo(ReplicationClientProgress const& progress)
+      : syncerId(progress.syncerId),
+        clientId(progress.clientId),
+        clientInfo(progress.clientInfo) {}
+
+  SyncerInfo(SyncerId syncerId, TRI_server_id_t clientId, std::string const& clientInfo)
+      : syncerId(syncerId), clientId(clientId), clientInfo(clientInfo) {}
+
+  SyncerId const syncerId;
+  TRI_server_id_t const clientId;
+  std::string const& clientInfo;
```

Comment:
Should this really be a reference?
That makes the SyncerInfo potentially unsafe or hard to use, if the original string becomes invalid and is still referenced here. I guess we can allow copying the string into the SyncerInfo for better safety & convenience.

Reply:
Well, I intended it only for formatting log messages, as in
```cpp
  LOG_TOPIC("c26ab", TRACE, Logger::REPLICATION)
      << "removing replication client entry for "
      << SyncerInfo{syncerId, clientId, clientInfo}
```
. But if you see it maybe being used in other cases, I've got no objections to using a string, I'll change it.